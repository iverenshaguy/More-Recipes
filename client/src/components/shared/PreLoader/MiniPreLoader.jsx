import React from 'react';
import Loader from 'react-loaders';
import 'loaders.css/src/animations/ball-beat.scss';
import './PreLoader.scss';

const MiniPreLoader = () => (<Loader type="ball-beat" />);

export default MiniPreLoader;
