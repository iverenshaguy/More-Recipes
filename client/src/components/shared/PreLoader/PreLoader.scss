@import '../../../assets/scss/modules';

.loader {
  &-hidden {
    display: none;
  }
  
  &-active {
    display: block;
  }

  .ball-beat > div {
    background-color: $light-green;
  }

  .ball-scale-multiple {
    margin: 30vh 0 25vh 46vw;
    height: 11vh;
    width: 11vh;

    & > div {
      left: 0;
    }
  }
}