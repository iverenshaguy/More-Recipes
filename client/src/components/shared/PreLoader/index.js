import React from 'react';
import Loader from 'react-loaders';

const PreLoader = () => (<Loader type="semi-circle-spin" style={{ transform: 'scale(0.5)' }} />);

export default PreLoader;
