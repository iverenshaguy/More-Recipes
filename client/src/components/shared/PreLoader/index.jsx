import React from 'react';
import Loader from 'react-loaders';
import './PreLoader.scss';

const PreLoader = () => (<Loader type="ball-scale-multiple" />);

export default PreLoader;
