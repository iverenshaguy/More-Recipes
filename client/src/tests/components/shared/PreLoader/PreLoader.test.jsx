import React from 'react';
import PreLoader, { MiniPreLoader } from '../../../../components/shared/PreLoader';

describe('PreLoaders', () => {
  it('renders correctly', () => {
    const mountedWrapper = mount(<PreLoader />);
    expect(toJson(mountedWrapper)).toMatchSnapshot();
    mountedWrapper.unmount();
  });

  it('renders MiniPreLoader correctly', () => {
    const mountedWrapper = mount(<MiniPreLoader />);
    expect(toJson(mountedWrapper)).toMatchSnapshot();
    mountedWrapper.unmount();
  });
});
