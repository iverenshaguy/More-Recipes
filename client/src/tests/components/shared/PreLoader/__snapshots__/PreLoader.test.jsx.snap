// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreLoaders renders MiniPreLoader correctly 1`] = `
<MiniPreLoader>
  <Loader
    active={true}
    type="ball-beat"
  >
    <div
      className="loader loader-undefined loader-active"
    >
      <div
        className="loader-inner ball-beat"
      >
        <div
          key="0"
          style={Object {}}
        />
        <div
          key="1"
          style={Object {}}
        />
        <div
          key="2"
          style={Object {}}
        />
      </div>
    </div>
  </Loader>
</MiniPreLoader>
`;

exports[`PreLoaders renders correctly 1`] = `
<PreLoader>
  <Loader
    active={true}
    type="ball-scale-multiple"
  >
    <div
      className="loader loader-undefined loader-active"
    >
      <div
        className="loader-inner ball-scale-multiple"
      >
        <div
          key="0"
          style={Object {}}
        />
        <div
          key="1"
          style={Object {}}
        />
        <div
          key="2"
          style={Object {}}
        />
      </div>
    </div>
  </Loader>
</PreLoader>
`;
